src/libmongoc/tests/test-mongoc-collection.c

```diff
@@ -3946,14 +3921,14 @@ _test_insert_validate (insert_fn_t insert_fn)
    mongoc_client_set_error_api (client, 2);
    collection = get_test_collection (client, "test_insert_validate");
 
-   BSON_ASSERT (!insert_fn (collection, tmp_bson ("{'$': 1}"), NULL, &error));
+   BSON_ASSERT (!insert_fn (collection, tmp_bson ("{'': 1}"), NULL, &error));
    ASSERT_ERROR_CONTAINS (error,
                           MONGOC_ERROR_COMMAND,
                           MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "invalid document");
+                          "empty key");
 
    BSON_ASSERT (!insert_fn (collection,
-                            tmp_bson ("{'$': 1}"),
+                            tmp_bson ("{'_id': {'$a': 1}}"),
```

Comment:
A dollar-prefixed field within an `_id` is one of the few server-side validation errors we can still rely upon regardless of server version. This is also used in the spec tests themselves.